modules/python/examples/ukf-nonlinear-complex-example.py

```diff
@@ -0,0 +1,536 @@
+
+  # Left turn
+  lastLinearVelocity = cmds[len(cmds) -1][0]
+  leftTurnCmds = generateTurnCommands(lastLinearVelocity, 0, 2, 15)
+  cmds.extend(leftTurnCmds)
+  for i in range(100):
+    cmds.append(cmds[len(cmds) -1])
+
+  # Right turn
+  lastLinearVelocity = cmds[len(cmds) -1][0]
+  rightTurnCmds = generateTurnCommands(lastLinearVelocity, 2, -2, 15);
+  cmds.extend(rightTurnCmds)
+  for i in range(200):
+    cmds.append(cmds[len(cmds) -1])
+
+  # Left  turn again
+  lastLinearVelocity = cmds[len(cmds) -1][0]
+  leftTurnCmds = generateTurnCommands(lastLinearVelocity, -2, 0, 15)
+  cmds.extend(leftTurnCmds)
+  for i in range(150):
+    cmds.append(cmds[len(cmds) -1])
+
+  lastLinearVelocity = cmds[len(cmds) -1][0]
+  leftTurnCmds = generateTurnCommands(lastLinearVelocity, 0, 1, 25)
+  cmds.extend(leftTurnCmds)
+  for i in range(150):
+    cmds.append(cmds[len(cmds) -1])
+
+  return cmds
+
+class vpBicycleModel:
+  """
+  @brief Class that approximates a 4-wheel robot unp.sing a bicycle model.
+  """
+  def __init__(self, w: float):
+    """
+    @brief Construct a new vpBicycleModel object.
+
+    @param w The length of the wheelbase.
+    """
+    self.m_w = w # The length of the wheelbase.
+
+  def computeMotion(self, u: ColVector, x: ColVector, dt: float) -> ColVector:
+    """
+    @brief Models the effect of the command on the state model.
+
+    @param u The commands. u[0] = velocity ; u[1] = steeringAngle .
+    @param x The state model. x[0] = x ; x[1] = y ; x[2] = heading
+    @param dt The period.
+    @return ColVector The state model after applying the command.
+    """
+    heading = x[2]
+    vel = u[0]
+    steeringAngle = u[1]
+    distance = vel * dt
+
+    if (abs(steeringAngle) > 0.001):
+      # The robot is turning
+      beta = (distance / self.m_w) * np.tan(steeringAngle)
+      radius = self.m_w / np.tan(steeringAngle);
+      np.sinh = np.sin(heading)
+      np.sinhb = np.sin(heading + beta)
+      np.cosh = np.cos(heading)
+      np.coshb = np.cos(heading + beta)
+      motion = ColVector([
+        -radius * np.sinh + radius * np.sinhb,
+        radius * np.cosh - radius * np.coshb,
+        beta
+      ])
+      return motion
+    else:
+      # The robot is moving in straight line
+      motion = ColVector([
+        distance * np.cos(heading),
+        distance * np.sin(heading),
+        0.
+      ])
+      return motion
+
+  def move(self, u: ColVector, x: ColVector, dt: float) -> ColVector:
+    """
+    @brief Models the effect of the command on the state model.
+
+    @param u The commands. u[0] = velocity ; u[1] = steeringAngle .
+    @param x The state model. x[0] = x ; x[1] = y ; x[2] = heading
+    @param dt The period.
+    @return ColVector The state model after applying the command.
+    """
+    motion = self.computeMotion(u, x, dt)
+    newX = x + motion
+    normalizedAngle = normalizeAngle(newX[2])
+    return ColVector([newX[0], newX[1], normalizedAngle])
+
+class vpLandmarkMeasurements:
```

Comment:
vpLandmarkMeasurements => Landmarkmeasurements

Reply:
Done in e885c41ad